### PR TITLE
Add release notes for 2.8.3

### PR DESCRIPTION
**Pull request by peytonrunyan**
[Rendered](https://github.com/PrefectHQ/prefect/blob/release-notes-2.8.3/RELEASE-NOTES.md)

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [ ] This pull request includes tests or only affects documentation.
- [ ] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`

### CODE REVIEW

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -1,4 +1,85 @@
 # Prefect Release Notes
+## Release 2.8.3
+
+### `on_completion` and `on_failure` hooks for flows and tasks 🎉
+With this release you can now add client-side hooks to your flows and tasks that will be called when the flow or task enters either a `Completed` or `Failed` state. This is great for any case where you might want to execute code without involvement of the Prefect API. 
+
+Both flows and tasks will accept the arguments `on_completion` and `on_failure` which take a list of callables. These callables will need to accept 3 arguments:
```
Comment (tpdorsey) [inline RELEASE-NOTES.md:7]:
```suggestion
Both flows and tasks will accept the arguments `on_completion` and `on_failure` that take a list of callables. These callables will need to accept 3 arguments:
```

File: RELEASE-NOTES.md [hunk 2]
```diff
@@ -1,4 +1,85 @@
...
+if __name__ == "__main__":
+    my_flow()
+```
+### Enhancements
+- Update `quote` handling in input resolution to skip descending into the quoted expression — https://github.com/PrefectHQ/prefect/pull/8576
+- Add light and dark mode color and contrast enhancements — https://github.com/PrefectHQ/prefect/pull/8629
+
+### Fixes
+- Fix `Task.map` type hint for async tasks — https://github.com/PrefectHQ/prefect/pull/8607
+- Update docker container name sanitization to handle "ce" and "ee" — https://github.com/PrefectHQ/prefect/pull/8588
```
Comment (tpdorsey) [inline RELEASE-NOTES.md:62]:
```suggestion
- Update Docker container name sanitization to handle "ce" and "ee" — https://github.com/PrefectHQ/prefect/pull/8588
```

File: RELEASE-NOTES.md [hunk 3]
```diff
@@ -1,4 +1,85 @@
...
+- Fix Kubernetes job watch timeout when streaming logs — https://github.com/PrefectHQ/prefect/pull/8618
```
Comment (tpdorsey) [inline RELEASE-NOTES.md:63]:
```suggestion
- Fix Kubernetes Job watch timeout when streaming logs — https://github.com/PrefectHQ/prefect/pull/8618
```

File: RELEASE-NOTES.md [hunk 4]
```diff
@@ -1,4 +1,85 @@
...
+- Fix flow run filters date range not working properly — https://github.com/PrefectHQ/prefect/pull/8616
```
Comment (tpdorsey) [inline RELEASE-NOTES.md:64]: Feels like this needs some rework to clarify what was done.

File: RELEASE-NOTES.md [hunk 5]
```diff
@@ -1,4 +1,85 @@
...
+- Fix Kubernetes log stream when using multiple containers in job — https://github.com/PrefectHQ/prefect/pull/8430
```
Comment (tpdorsey) [inline RELEASE-NOTES.md:65]:
```suggestion
- Fix Kubernetes log stream when using multiple containers in Job — https://github.com/PrefectHQ/prefect/pull/8430
```

File: RELEASE-NOTES.md [hunk 6]
```diff
@@ -1,4 +1,85 @@
...
+
+### Experimental
+- Update worker variable typing for clearer display in the UI — https://github.com/PrefectHQ/prefect/pull/8613
+- Update `BaseWorker` to ignore flow runs with associated storage block — https://github.com/PrefectHQ/prefect/pull/8619
+- Add experimental artifacts API — https://github.com/PrefectHQ/prefect/pull/8404
+
+### Documentation
+- Add Resume Flow Run Through UI — https://github.com/PrefectHQ/prefect/pull/8621
+- Add `prefect-sifflet` to Collections catalog — https://github.com/PrefectHQ/prefect/pull/8599
```
Comment (tpdorsey) [inline RELEASE-NOTES.md:74]:
```suggestion
- Add [`prefect-sifflet`](https://siffletapp.github.io/prefect-sifflet/) to Collections catalog — https://github.com/PrefectHQ/prefect/pull/8599
```

File: docs/overrides/main.html [hunk 7]
```diff
@@ -1,10 +1,10 @@
 {% extends "base.html" %}
 {% block announce %}
 <div style="color: white;">
-  <a href="https://github.com/PrefectHQ/prefect/blob/main/RELEASE-NOTES.md#release-281" style="color: #07e798;">
-    <strong>Prefect 2.8.1 is live!</strong>
-  </a> We've replaced "Orion" in commands, components, and the API with more explicit nomenclature, plus 
-  <a href="https://github.com/PrefectHQ/prefect/blob/main/RELEASE-NOTES.md#release-281" style="color: #07e798;">
+  <a href="https://github.com/PrefectHQ/prefect/blob/main/RELEASE-NOTES.md#release-283" style="color: #07e798;">
+    <strong>Prefect 2.8.3 is live!</strong>
+  </a> Adding support for client-side state change hooks, plus
```
Comment (tpdorsey) [inline docs/overrides/main.html:6]:
```suggestion
  </a> Adding support for client-side state change hooks, new Sifflet Collection, plus
```

File: RELEASE-NOTES.md [hunk 8]
```diff
@@ -1,4 +1,85 @@
...
+@flow
+def my_flow():
+    my_task.submit()
+
+if __name__ == "__main__":
+    my_flow()
+```
+### Enhancements
+- Update `quote` handling in input resolution to skip descending into the quoted expression — https://github.com/PrefectHQ/prefect/pull/8576
+- Add light and dark mode color and contrast enhancements — https://github.com/PrefectHQ/prefect/pull/8629
```
Comment (serinamarie) [inline RELEASE-NOTES.md:58]:
```suggestion
- Add light and dark mode color and contrast enhancements to UI — https://github.com/PrefectHQ/prefect/pull/8629
```

File: RELEASE-NOTES.md [hunk 9]
```diff
@@ -1,4 +1,85 @@
 # Prefect Release Notes
+## Release 2.8.3
```
Comment (madkinsz) [inline RELEASE-NOTES.md:2]:
```suggestion
# Prefect Release Notes

## Release 2.8.3
```

File: RELEASE-NOTES.md [hunk 10]
```diff
@@ -1,4 +1,85 @@
...
+
+### `on_completion` and `on_failure` hooks for flows and tasks
+With this release you can now add client-side hooks to your flows and tasks that will be called when the flow or task enters either a `Completed` or `Failed` state. This is great for any case where you might want to execute code without involvement of the Prefect API.
```
Comment (madkinsz) [inline RELEASE-NOTES.md:5]:
```suggestion
With this release you can now add client-side hooks that will be called when your flow or task enters a `Completed` or `Failed` state. This is great for any case where you want to execute code without involvement of the Prefect API. 
```

File: RELEASE-NOTES.md [hunk 11]
```diff
@@ -1,4 +1,85 @@
 # Prefect Release Notes
+## Release 2.8.3
+
+### `on_completion` and `on_failure` hooks for flows and tasks
+With this release you can now add client-side hooks to your flows and tasks that will be called when the flow or task enters either a `Completed` or `Failed` state. This is great for any case where you might want to execute code without involvement of the Prefect API. 
+
+Both flows and tasks will accept the arguments `on_completion` and `on_failure` that take a list of callables. These callables will need to accept 3 arguments:
```
Comment (madkinsz) [inline RELEASE-NOTES.md:7]:
```suggestion
Both flows and tasks include `on_completion` and `on_failure` options where a list of callable hooks can be provided. The callable will receive three arguments:
```

File: RELEASE-NOTES.md [hunk 12]
```diff
@@ -1,4 +1,85 @@
...
+- `flow`, `flow_run`, and `state` in the case of a flow hook
+- `task`, `task_run`, and `state` in the case of a task hook
+
+Here is an example showing how completion hooks can be added to a flow and a task:
```
Comment (madkinsz) [inline RELEASE-NOTES.md:11]:
```suggestion
For example, here we add completion hooks to a flow and a task:
```

File: RELEASE-NOTES.md [hunk 13]
```diff
@@ -1,4 +1,85 @@
...
+    print("This is the task!")
+
+@flow(on_completion=[my_completion_flow_hook])
+def my_flow():
+    my_task()
+
+if __name__ == "__main__":
+    my_flow()
+```
+Here is an example showing how failure hooks work. It's worth noting that you can supply both `on_completion` and `on_failure` hooks to a flow or task. Only the hooks that are relevant to the state of the flow or task will be called.
```
Comment (madkinsz) [inline RELEASE-NOTES.md:35]:
```suggestion
Next, we'll include a failure hook as well. It's worth noting that you can supply both `on_completion` and `on_failure` hooks to a flow or task. Only the hooks that are relevant to the final state of the flow or task will be called.
```

File: RELEASE-NOTES.md [hunk 14]
```diff
@@ -1,4 +1,85 @@
...
+def my_task():
+    print("This is the task!")
+
+@flow(on_completion=[my_completion_flow_hook])
+def my_flow():
+    my_task()
+
+if __name__ == "__main__":
+    my_flow()
+```
```
Comment (madkinsz) [inline RELEASE-NOTES.md:34]: Try to include spacing between code fences and content for readability.

File: RELEASE-NOTES.md [hunk 15]
```diff
@@ -1,4 +1,85 @@
...
+- Fix `Task.map` type hint for type-checker compatibility with async tasks — https://github.com/PrefectHQ/prefect/pull/8607
+- Update Docker container name sanitization to handle "ce" and "ee" when checking Docker version — https://github.com/PrefectHQ/prefect/pull/8588
+- Fix Kubernetes Job watch timeout behavior when streaming logs — https://github.com/PrefectHQ/prefect/pull/8618
+- Fix date range filter selection on the flow runs UI page — https://github.com/PrefectHQ/prefect/pull/8616
+- Fix Kubernetes not streaming logs when using multiple containers in job — https://github.com/PrefectHQ/prefect/pull/8430
+
+### Experimental
+- Update worker variable typing for clearer display in the UI — https://github.com/PrefectHQ/prefect/pull/8613
+- Update `BaseWorker` to ignore flow runs with associated storage block — https://github.com/PrefectHQ/prefect/pull/8619
+- Add experimental artifacts API — https://github.com/PrefectHQ/prefect/pull/8404
```
Comment (madkinsz) [inline RELEASE-NOTES.md:70]:
```suggestion
- Add experimental API for artifacts — https://github.com/PrefectHQ/prefect/pull/8404
```

File: RELEASE-NOTES.md [hunk 16]
```diff
@@ -1,4 +1,85 @@
...
+
+### Documentation
+- Add Resume Flow Run Through UI — https://github.com/PrefectHQ/prefect/pull/8621
```
Comment (madkinsz) [inline RELEASE-NOTES.md:73]:
```suggestion
- Add documentation for resuming a flow run via the UI — https://github.com/PrefectHQ/prefect/pull/8621
```

File: RELEASE-NOTES.md (hunk 4 above)
Comment (peytonrunyan) [reply RELEASE-NOTES.md]: Could you give it a look now? The Kubernetes job watch timeout involves a few changes that I don't think are easy to capture in a single sentence, but the others I think are more clear.

File: RELEASE-NOTES.md (hunk 14 above)
Comment (peytonrunyan) [reply RELEASE-NOTES.md:34]: Cool - I wasn't certain where the line between proper spacing and not making the example part of the release notes too long was.

Comment (madkinsz) [reply RELEASE-NOTES.md:34]: I don't think it matters how long the markdown is. I try not to have excessive spacing in the code itself though.

**Review by madkinsz — APPROVED** [review]

File: RELEASE-NOTES.md [hunk 17]
```diff
@@ -1,5 +1,87 @@
...
+### Enhancements
+- Update `quote` handling in input resolution to skip descending into the quoted expression — https://github.com/PrefectHQ/prefect/pull/8576
+- Add light and dark mode color and contrast enhancements to UI — https://github.com/PrefectHQ/prefect/pull/8629
+
+### Fixes
+- Fix `Task.map` type hint for type-checker compatibility with async tasks — https://github.com/PrefectHQ/prefect/pull/8607
+- Update Docker container name sanitization to handle "ce" and "ee" when checking Docker version — https://github.com/PrefectHQ/prefect/pull/8588
+- Fix Kubernetes Job watch timeout behavior when streaming logs — https://github.com/PrefectHQ/prefect/pull/8618
+- Fix date range filter selection on the flow runs UI page — https://github.com/PrefectHQ/prefect/pull/8616
+- Fix Kubernetes not streaming logs when using multiple containers in job — https://github.com/PrefectHQ/prefect/pull/8430
```
Comment (tpdorsey) [inline RELEASE-NOTES.md]:
```suggestion
- Fix Kubernetes not streaming logs when using multiple containers in Job — https://github.com/PrefectHQ/prefect/pull/8430
```

**Pull request merged**